rc/filetype/scheme.kak

```diff
@@ -45,64 +45,88 @@ add-highlighter shared/scheme/comment-block region "#\|" "\|#" fill comment
 add-highlighter shared/scheme/quoted-form region -recurse "\(" "'\(" "\)" fill variable
 
 add-highlighter shared/scheme/code/ regex (#t|#f) 0:value
-add-highlighter shared/scheme/code/ regex \b[0-9]+\.[0-9]*\b 0:value
+
+# Numbers
+add-highlighter shared/scheme/code/ regex '(?:#[bB]#[eiEI]|#[eiEI]#[bB]|#[bB])[01]+' 0:value
+add-highlighter shared/scheme/code/ regex '(?:#[oO]#[eiEI]|#[eiEI]#[oO]|#[oO])[0-7]+' 0:value
+add-highlighter shared/scheme/code/ regex '(?:#[dD](?:#[eiEI])?|#[eiEI]#[dD]|#[eiEI])(?:\d+(?:\.\d*)?|\.\d+)(?:[esfdlESFDL][-+]?\d+)?' 0:value
+add-highlighter shared/scheme/code/ regex '(?:#[xX]#[eiEI]|#[eiEI]#[xX]|#[xX])[0-9a-fA-F]+' 0:value
```

Comment:
Since the highlighter applies to the entire matched text, maybe the pattern would be more readable with simple parenthesis group, as opposed to make them all unmatching with `?:`?

Reply:
Hmm, I guess that must be an old habit... Maybe it was inefficient for some old regex engine?  Updated.